py/desitarget/mock/io.py

```diff
@@ -642,7 +642,7 @@ def read_gaussianfield(mock_dir_name, target_name, rand=None, bricksize=0.25,
             lyafiles = lyafiles_qso
             lyahdu = lyahdu_qso
             templatetype = templatetype_qso
-            templatesubtype = templatesubtype_qso
+            templatesubtype = templatesubtype
         else:
```

Comment:
This is definitely a bug but it should be fixed on line 611, not here.  This line (645) should still be `templatesubtype = templatesubtype_qso` but then line 611 should be `templatesubtype_qso = np.repeat('', nobj_qso)`.

Reply:
All right. I will make that change.